osgar/drivers/lidar.py

```diff
@@ -0,0 +1,70 @@
+"""
+  SICK Tim571 LIDAR Driver
+"""
+
+from threading import Thread
+
+from osgar.drivers.bus import BusShutdownException
+
+
+STX = b'\x02'
+ETX = b'\x03'
+
+
+class Lidar(Thread):
+    def __init__(self, config, bus):
+        Thread.__init__(self)
+        self.setDaemon(True)
```

Comment:
Wasn't there some problem with the Lidar when it was not properly shut down? I vaguely remember that the Lidar would not return any data for 20s after improper shutdown. Is that correct? If so, is the issue handled here? I believe daemon threads are just killed when the process ends.

Reply:
Yes, there was a problem (LMS100 on John Deere) - thanks for reminder. The issue there was that due to energy saving we switched laser off (to some "stand-by" mode) at the end of the run. But sometimes this message was not properly communicated with the sensor. The next run had 20s "gap" as the sensor was booting up, sometimes. Also this 20s was LMS100 sending data, but they did not change and corresponded to the last valid reading (?).

I am not sure how it is with TIM571, but for now I would just not switch the sensor off.

Reply:
> Yes, there was a problem (LMS100 on John Deere) - thanks for reminder.

Oh, if this is only for Tim571, I think it shouldn't be called `lidar.py`. It sets the wrong expectations. 

Anyway, IMO there should be no daemon threads in the system. All threads should end correctly. Is there any reason to have daemon threads?